### PR TITLE
optimization use small blocks up to 64KB

### PR DESCRIPTION
**Pull request by klauspost**
## Description

From https://github.com/minio/minio/discussions/17103 - files up to 1MB would be read in 32KB blocks. Set that threshold to 64KB, so at most there are 2 reads.

Remove the unneeded wrapper in readAllData when we are not using O_DIRECT - we are reading directly into a single buffer, so there is no need for an intermediate buffer. This will save a memcopy in most cases.

Either way, we now quickly reject on unsupported platforms and avoid the copying there.

Bonus: `ReadFileWithFileInfo` will now also use `NO_ATIME` flag for reads.
 
## How to test this PR?

Picked up by regular tests reliably.

## Types of changes
- [x] Optimization (provides speedup with no functional changes)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: internal/ioutil/read_file.go [hunk 1]
```diff
@@ -62,6 +62,23 @@ func ReadFileWithFileInfo(name string) ([]byte, fs.FileInfo, error) {
 //
 // passes NOATIME flag for reads on Unix systems to avoid atime updates.
 func ReadFile(name string) ([]byte, error) {
+	if !disk.ODirectPlatform {
+		// Don't wrap with un-needed buffer.
+		// Don't use os.ReadFile, since it doesn't pass NO_ATIME when present.
+		f, err := OpenFileDirectIO(name, readMode, 0o666)
```
Comment (vadmeste) [inline internal/ioutil/read_file.go]: Shouldn't we use OpenFileDirectIO since disk.ODirectPlatform == false in this case ?

Comment (klauspost) [reply internal/ioutil/read_file.go]: Yeah. It falls back on the implementation, but you are completely correct.

**Review by harshavardhana — APPROVED** [review]

**Review by krisis — APPROVED** [review]

**Comment by tdsn0007** [comment]
Hi, this optimization will help to read 1MB files (like veeam files)?

**Comment by klauspost** [comment]
@tdsn0007 A tiny bit.

**Pull request merged**